### ISSUE TITLE
autoaugment

### ISSUE THREAD

**Issue by mbenami** [tensorflow/models#6144]
Hi.
Is it possible to release the code for training autoaugment policies?
Thanks

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by ymodak** [comment]
Please take a look at https://github.com/tensorflow/models/tree/master/research/autoaugment

**Comment by mbenami** [comment]
hi @ymodak thanks for the respond 
the code in the repo you mention is related to the train with the polices found in the paper 
I have a different dataset with different augmentation 
I would like to train the reinforcement network to find the best augmentation fits my dataset

**Comment by fnzhan** [comment]
I am also interested in the code using reinforcement learning to find the policies, have sent emails, while no response.

**Comment by mrottmann** [comment]
Hi, I used the autoaugment code and I am not able to reproduce the result for cifar10 and wide ResNet from the paper. I'm using openSUSE Leap 42.3, python3.4, tf-gpu 1.13.1 installed via pip, a Quadro P6000 (24GB), libcublas.so.10.0, CUDA Version 10.0.130, cuDNN 7.4, gcc 4.8.5.

I get about 96.07% test accuracy (and 97.69% train acc), I did not touch any of the parameters and used the run command suggested on the github page. Only change I made was adjusting the pickle calls to be suitable for python3.

**Comment by Ramay7** [comment]
@fnzhan Hi, I have the same interest as you. I want to know how the reward (accuracy on validation set) will guide the training in detail. Because I think the reward signal is too weak and may change only in small scale. Have you find some code about reinforcement learning part ? If so, is it convenient to share?

Thanks!

**Comment by fnzhan** [comment]
@Ramay7 Hi, the author said that the search model is exactly the same as the previous neural architecture search paper https://arxiv.org/abs/1611.01578 whose code can be found here  https://github.com/titu1994/neural-architecture-search. I didn't try to modify this code. I feel the search code of autoaugment will not be released.

**Comment by Ramay7** [comment]
> @Ramay7 Hi, the author said that the search model is exactly the same as the previous neural architecture search paper https://arxiv.org/abs/1611.01578 whose code can be found here https://github.com/titu1994/neural-architecture-search. I didn't try to modify this code. I feel the search code of autoaugment will not be released.

OK. Thanks for your reply:) I will read the paper and code soon.

**Comment by BarretZoph** [comment]
> Hi, I used the autoaugment code and I am not able to reproduce the result for cifar10 and wide ResNet from the paper. I'm using openSUSE Leap 42.3, python3.4, tf-gpu 1.13.1 installed via pip, a Quadro P6000 (24GB), libcublas.so.10.0, CUDA Version 10.0.130, cuDNN 7.4, gcc 4.8.5.
> 
> I get about 96.07% test accuracy (and 97.69% train acc), I did not touch any of the parameters and used the run command suggested on the github page. Only change I made was adjusting the pickle calls to be suitable for python3.

Right now it appears that the code only works with python2 and not python3. We are looking more into this.

**Comment by mrottmann** [comment]
> > Hi, I used the autoaugment code and I am not able to reproduce the result for cifar10 and wide ResNet from the paper. I'm using openSUSE Leap 42.3, python3.4, tf-gpu 1.13.1 installed via pip, a Quadro P6000 (24GB), libcublas.so.10.0, CUDA Version 10.0.130, cuDNN 7.4, gcc 4.8.5.
> > I get about 96.07% test accuracy (and 97.69% train acc), I did not touch any of the parameters and used the run command suggested on the github page. Only change I made was adjusting the pickle calls to be suitable for python3.
> 
> Right now it appears that the code only works with python2 and not python3. We are looking more into this.

Thanks for your reply and the hint. I'm now running the code with python2.7 and I get 2.90% test error which is now much closer to the results in your paper. From the paper it does not seem like you averaged the results over several runs. From your experience, would you say that 2.90% is still within the expected fluctuation range?
